<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>699</width>
    <height>536</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>LogicCommPort</string>
  </property>
  <property name="autoFillBackground">
   <bool>false</bool>
  </property>
  <property name="locale">
   <locale language="English" country="UnitedStates"/>
  </property>
  <property name="unifiedTitleAndToolBarOnMac">
   <bool>false</bool>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="QGroupBox" name="groupBox_2">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
        <horstretch>5</horstretch>
        <verstretch>1</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>246</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="title">
       <string>OUTPUT DATA</string>
      </property>
      <widget class="QTextEdit" name="textEd_out">
       <property name="geometry">
        <rect>
         <x>17</x>
         <y>125</y>
         <width>447</width>
         <height>78</height>
        </rect>
       </property>
      </widget>
      <widget class="QWidget" name="layoutWidget">
       <property name="geometry">
        <rect>
         <x>18</x>
         <y>96</y>
         <width>444</width>
         <height>24</height>
        </rect>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QLabel" name="label">
          <property name="text">
           <string>Data for send</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>18</width>
            <height>14</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QRadioButton" name="rbt_Rawout">
          <property name="text">
           <string>Raw</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="rbt_Decout">
          <property name="text">
           <string>Dec</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="rbt_ASCIIout">
          <property name="text">
           <string>ASCII</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="rbt_BINout">
          <property name="text">
           <string>BIN</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="rbt_HEXout">
          <property name="text">
           <string>HEX</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="layoutWidget">
       <property name="geometry">
        <rect>
         <x>18</x>
         <y>26</y>
         <width>441</width>
         <height>29</height>
        </rect>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout_11">
        <item>
         <widget class="QLabel" name="label_7">
          <property name="text">
           <string>Period</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="spinBox_Period">
          <property name="suffix">
           <string> ms</string>
          </property>
          <property name="maximum">
           <number>10000</number>
          </property>
          <property name="singleStep">
           <number>10</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QProgressBar" name="progressBar">
          <property name="value">
           <number>24</number>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="layoutWidget">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>60</y>
         <width>441</width>
         <height>29</height>
        </rect>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QPushButton" name="bt_send">
          <property name="statusTip">
           <string>Send data</string>
          </property>
          <property name="text">
           <string>SEND</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="bt_stopsend">
          <property name="statusTip">
           <string>Stop send</string>
          </property>
          <property name="text">
           <string>STOP</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer_2">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>218</width>
            <height>24</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
      <widget class="QPushButton" name="bt_open">
       <property name="geometry">
        <rect>
         <x>18</x>
         <y>210</y>
         <width>85</width>
         <height>27</height>
        </rect>
       </property>
       <property name="text">
        <string>Open</string>
       </property>
      </widget>
      <widget class="QPushButton" name="bt_save">
       <property name="geometry">
        <rect>
         <x>109</x>
         <y>210</y>
         <width>85</width>
         <height>27</height>
        </rect>
       </property>
       <property name="text">
        <string>Save</string>
       </property>
      </widget>
      <zorder>layoutWidget</zorder>
      <zorder>layoutWidget</zorder>
      <zorder>layoutWidget</zorder>
      <zorder>bt_open</zorder>
      <zorder>bt_save</zorder>
      <zorder>textEd_out</zorder>
     </widget>
    </item>
    <item row="1" column="1">
     <widget class="QDockWidget" name="dockWidgetmessage">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
        <horstretch>2</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="maximumSize">
       <size>
        <width>524287</width>
        <height>524287</height>
       </size>
      </property>
      <property name="autoFillBackground">
       <bool>false</bool>
      </property>
      <property name="windowTitle">
       <string>Messages</string>
      </property>
      <widget class="QWidget" name="dockWidgetContents">
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="0" column="0">
         <widget class="QTextBrowser" name="textBr_mess"/>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QGroupBox" name="groupBox">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
        <horstretch>2</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="title">
       <string>PORT SETTINGS</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_5">
         <item>
          <widget class="QLabel" name="label_2">
           <property name="text">
            <string>Port</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="comboBox_port"/>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_7">
         <item>
          <widget class="QPushButton" name="bt_openport">
           <property name="statusTip">
            <string>Open port</string>
           </property>
           <property name="text">
            <string>OPEN</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="bt_closeport">
           <property name="statusTip">
            <string>Close port</string>
           </property>
           <property name="text">
            <string>CLOSE</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>21</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_6">
         <item>
          <widget class="QLabel" name="label_3">
           <property name="text">
            <string>Baudrate</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="comboBox_baudrate"/>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_8">
         <item>
          <widget class="QLabel" name="label_5">
           <property name="text">
            <string>Parity</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="comboBox_parity"/>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_9">
         <item>
          <widget class="QLabel" name="label_6">
           <property name="text">
            <string>Flowcontrol</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="comboBox_flcntrl"/>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_10">
         <item>
          <widget class="QPushButton" name="bt_apply">
           <property name="statusTip">
            <string>Apply port settings</string>
           </property>
           <property name="text">
            <string>APPLY</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="bt_cancel">
           <property name="statusTip">
            <string>Cancel new settings </string>
           </property>
           <property name="text">
            <string>CANCEL</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QGroupBox" name="groupBox_3">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
        <horstretch>5</horstretch>
        <verstretch>2</verstretch>
       </sizepolicy>
      </property>
      <property name="title">
       <string>INPUT DATA</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <item>
          <widget class="QPushButton" name="bt_clear">
           <property name="statusTip">
            <string>Clear input data</string>
           </property>
           <property name="text">
            <string>CLEAR</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer_4">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>308</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_4">
         <item>
          <widget class="QLabel" name="label_4">
           <property name="text">
            <string>Format:</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer_1">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>58</width>
             <height>17</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QRadioButton" name="rbt_Decin">
           <property name="text">
            <string>Decimal</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="rbt_ASCIIin">
           <property name="text">
            <string>ASCII</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="rbt_NMEAin">
           <property name="text">
            <string>NMEA</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="rbt_BINin">
           <property name="text">
            <string>BIN</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="rbt_HEXin">
           <property name="text">
            <string>HEX</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QTextBrowser" name="textBr_inp">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>699</width>
     <height>25</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar">
   <property name="sizeGripEnabled">
    <bool>true</bool>
   </property>
  </widget>
 </widget>
 <tabstops>
  <tabstop>comboBox_port</tabstop>
  <tabstop>bt_openport</tabstop>
  <tabstop>bt_closeport</tabstop>
  <tabstop>comboBox_baudrate</tabstop>
  <tabstop>comboBox_parity</tabstop>
  <tabstop>comboBox_flcntrl</tabstop>
  <tabstop>bt_apply</tabstop>
  <tabstop>bt_cancel</tabstop>
  <tabstop>textEd_out</tabstop>
  <tabstop>bt_clear</tabstop>
  <tabstop>textBr_inp</tabstop>
  <tabstop>textBr_mess</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
